### PR TITLE
Added Copy to clipboard and Save text file of the bookmarks.

### PR DESCRIPTION
**Pull request by EroticRide**
## Proposed changes

Added 2 buttons in the ending screen: 

1. **Copy to Clipboard**: copies all your links of your bookmarks into your device's clipboard
2. **Save text file**: downloads a txt file to your device with all the bookmarks.

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe):

### CODE REVIEW

File: src/features/end/routes/End.tsx [hunk 1]
```diff
@@ -33,6 +35,46 @@ export function End() {
...
+    const element = document.createElement("a");
+    const file = new Blob([bmList], {
+      type: "text/plain"
+    });
+    element.href = URL.createObjectURL(file);
+
+    const today = new Date();
+    const dateStr = today.getFullYear() + "_" + today.getMonth() + "_" + today.getDate() + " " + today.toLocaleTimeString()
+
+    element.download = "fapInstructorBookmarks_" + dateStr + ".txt";
```
Comment (thefapinstructor) [inline src/features/end/routes/End.tsx:72]:
```suggestion
    element.download = `fapinstructor_bookmarks_${format(Date.now(), "yyyy_MM_dd")}.txt`
```

**Review by thefapinstructor — CHANGES_REQUESTED** [review]
Thanks for your PR! :pray: :tada:.  I left a couple of small refactor suggestions, once ready I can merge!

File: src/features/end/routes/End.tsx [hunk 2]
```diff
@@ -33,6 +35,46 @@ export function End() {
...
+
+    store.game.bookmarks.forEach(bookmark => {
+      if (bmList !==""){
+        bmList = bmList + "\n" + bookmark.href;
+      } else {
+        bmList = bookmark.href;
+      }     
+    });
+    await navigator.clipboard.writeText(bmList);
+    alert('Bookmarks copied');
```
[the remark this answers is not in the document]
Comment (EroticRide) [reply src/features/end/routes/End.tsx]: Just changed that in my most recent commit.

**Comment by EroticRide** [comment]
Just changed the alert box in my most recent commit. 

However, the other two changes you suggested don't seem to work.

1. The way to collect the bookmarks you suggested returns an array of objects. So when you copy/download shows as follows: 
Bookmark1
2. The function you suggested to use for the date formatting doesn't exist in the project and won't compile because of it. Also In my example the time of the day is also included. (For the people that may want to play multiple games a day 😅  )

**Comment by thefapinstructor** [comment]
> Just changed the alert box in my most recent commit.
> 
> However, the other two changes you suggested don't seem to work.
> 
> 1. The way to collect the bookmarks you suggested returns an array of objects. So when you copy/download shows as follows:
>    Bookmark1
> 2. The function you suggested to use for the date formatting doesn't exist in the project and won't compile because of it. Also In my example the time of the day is also included. (For the people that may want to play multiple games a day sweat_smile  )

1. I forgot that bookmarks is an object.  You should be able to do something like
```ts
await navigator.clipboard.writeText(store.game.bookmarks.map(bookmark => bookmark.href).join("\n")
```
2. `format` is exported from the wonderful library [date-fns](https://date-fns.org/)
```
import format from "date-fns/format";
```

**Comment by EroticRide** [comment]
> 

All done, thanks for the help 👍

**Review by thefapinstructor — APPROVED** [review]
LGTM! Thanks for your contribution :heart:

**Pull request merged**